cmd/up/stignore.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package up
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	initCMD "github.com/okteto/okteto/cmd/init"
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/linguist"
+	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+)
+
+func checkStignoreConfiguration(dev *model.Dev) error {
+	for _, folder := range dev.Syncs {
+		stignorePath := filepath.Join(folder.LocalPath, ".stignore")
+		gitPath := filepath.Join(folder.LocalPath, ".git")
+		if !model.FileExists(stignorePath) {
+			log.Infof("'.stignore' does not exist in folder '%s'", folder.LocalPath)
+			if err := askIfCreatingStignore(folder.LocalPath, stignorePath, gitPath); err != nil {
+				return err
+			}
+			continue
+		}
+
+		log.Infof("'.stignore' exists in folder '%s'", folder.LocalPath)
+		if !model.FileExists(gitPath) {
+			continue
+		}
+
+		if err := askIfUpdatingStignore(folder.LocalPath, stignorePath, gitPath); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func askIfCreatingStignore(folder, stignorePath, gitPath string) error {
+	log.Information("It is recommended to define a '.stignore' file to avoid synchronizing things like build artifacts, git metadata or third party libraries.")
+	answer, err := utils.AskYesNo("    Do you want to automatically create a '.stignore' file? [y/n] ")
+	if err != nil {
+		return fmt.Errorf("failed to ask for adding '.stignore' to '%s': %s", folder, err.Error())
+	}
+
+	if !answer {
+		stignoreContent := ""
+		if model.FileExists(gitPath) {
+			stignoreContent = ".git\n"
+		}
+		if err := ioutil.WriteFile(stignorePath, []byte(stignoreContent), 0644); err != nil {
```

Comment:
if the user chose no as the answer,  why do we still create the ignore file?

Reply:
It servers to goals:
- only ask once
- more discoverability of the feature

Reply:
but it goes against what you're asking the user. if they are asked "should I create the stignore file", they say no, and the file is still created they are going to think there's a bug. if you want to always create the file, then don't ask them (I don't think this is right approach btw).

Reply:
what do you suggest?

Reply:
if the user says no, we don't create the file, and don't ask them again for that repo.

Reply:
I understand the painpoint, but we need to find a way to educate the user rather than ignore their answers. I think we should explain why synchronizing .git is a bad idea, but if they want to do it, it's their choice.

Reply:
Right, but the implementation of "don't ask them again for that repo" must live in the repo. If not, every time a user clones the repo will get prompted, and that feels wrong because okteto has been already configured.
I will rephrase the question to make it clear.

Reply:
It did look like a rephrasing issue. I like it better now, but maybe we should also state the benefits of doing so. What about something like: 
```Okteto requires a '.stignore' file to intentionally ignore file patterns that help optimize the synchronization service.``` ?